frame/election-provider-support/src/lib.rs

```diff
@@ -338,10 +338,13 @@ pub trait SortedListProvider<AccountId> {
 		weight_of: Box<dyn Fn(&AccountId) -> VoteWeight>,
 	) -> u32;
 
-	/// Remove `maybe_count` number of items from the list. Returns the number of items actually
-	/// removed. WARNING: removes all items if `maybe_count` is `None`, which should never be done
-	/// in production settings because it can lead to an unbounded amount of storage accesses.
-	fn clear(maybe_count: Option<u32>) -> u32;
+	/// Remove all items from the list.
+	///
+	/// ## WARNING
+	///
+	/// This function should never be called in production settings because it can lead to an
+	/// unbounded amount of storage accesses.
+	fn clear();
```

Comment:
Maybe call this `unsafe_clear` then? To make the user aware of the problems?

Reply:
also in pallet-bags-list when implementing the function `SortedListProvider::regenerate` it calls `clear` inside. Should we also prepend unsafe to regenerate ? WDYT @kianenigma

Reply:
Yes we should

Reply:
addressed in https://github.com/paritytech/substrate/pull/10231/commits/fede8863a60477476ba519f9d3365bdd61f05076